File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Model/Pago.java
package com.example.BackendCitasMedicas.Model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Pago {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id_pago;
    private LocalDate fecha_pago;
    private int monto;
    private String metodo_pago;
    private String referencia;

    @ManyToOne
    @JoinColumn(name = "id_paciente")
    private Paciente paciente;
}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Model/Pago.kt
package com.example.frontcitasmedicas.Model

import java.time.LocalDate

data class Pago(
    val id_pago: Long = 0,
    val fecha_pago: String = "", // O `LocalDate`
    val monto: Int = 0,
    val metodo_pago: String = "",
    val referencia: String = "",
    val paciente: Paciente? = null
)
